src/traverse/node-path.js

```diff
@@ -34,9 +34,62 @@ class NodePath {
   }
 
   /**
-   * Inserts a node into a collection. By default child nodes
-   * are supposed to by under `expressions` property. An explicit
-   * property can be passed.
+   * sets a node into a children collection or the single child.
+   * By default child nodes are supposed to be under `expressions` property.
+   * An explicit property can be passed.
+   *
+   * @param Object node - a node to set into a collection or as single child
+   * @param number index - index at which to set
+   * @param string property - name of the collection or single property
+   */
+  setChild(node, index = null, property = null) {
+
+    if (index != null) {
+      if (!property) property = "expressions";
+      if (!this.node.hasOwnProperty(property)) {
+        throw new Error(
+          `Node of type ${this.node.type} doesn't have "${property}" collection.`
+        );
+      }
```

Comment:
this can be moved to a helper method, and reused in other methods (with leading `_` underscores "private" methods are defined in JS by convention)

```js
this._enforceCollectionProp(property);
```